### PR TITLE
[feature] Automate wheel builds on new releases

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Credits to @myleott, this is a litteral copy pasta

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/build_wheels.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Install cibuildwheel
+        run: |
+          python -m pip install cibuildwheel
+      - name: Build wheels for CPython
+        run: |
+          python -m cibuildwheel --output-dir dist
+        env:
+          CIBW_BUILD: "cp36-*64 cp37-*64 cp38-*64"
+          CIBW_MANYLINUX_X86_64_IMAGE: manylinux1
+          CIBW_BEFORE_BUILD: git submodule update --init --recursive && pip install .
```
Comment (myleott) [inline .github/workflows/build_wheels.yml:35]: If there’s no submodules you might be able to remove that part (but leave the pip install part)

File: .github/workflows/build_wheels.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          python-version: '3.7'
+
+      - name: Install cibuildwheel
+        run: |
+          python -m pip install cibuildwheel
+      - name: Build wheels for CPython
+        run: |
+          python -m cibuildwheel --output-dir dist
+        env:
+          CIBW_BUILD: "cp36-*64 cp37-*64 cp38-*64"
```
Comment (min-xu-ai) [inline .github/workflows/build_wheels.yml:33]: py39 please :-)

**Review by min-xu-ai — APPROVED** [review]
super

File: .github/workflows/build_wheels.yml (hunk 2 above)
Comment (blefaudeux) [reply .github/workflows/build_wheels.yml]: done !

File: pyproject.toml [hunk 3]
```diff
@@ -28,4 +28,4 @@ use_parentheses = true
 skip_glob = ["build/*", "stubs/*"]
 # Don't split "import" and "from".
 force_sort_within_sections = true
-known_third_party = ["benchmark_dataset", "datasets", "golden_configs", "helpers", "models", "numpy", "pytest", "recommonmark", "setuptools", "torch", "torch_pg", "torchtext", "torchvision"]
+known_third_party = ["benchmark_dataset", "dataclasses", "datasets", "golden_configs", "helpers", "models", "numpy", "pytest", "recommonmark", "setuptools", "torch", "torch_pg", "torchtext", "torchvision"]
```
Comment (msbaines) [inline pyproject.toml:31]: This doesn't seem right. dataclasses is builtin to Python.

**Pull request merged**